state/protocol/badger/mutator.go

```diff
@@ -601,12 +603,9 @@ func (m *Mutator) receiptExtend(candidate *flow.Block) error {
 			return fmt.Errorf("payload includes receipt for block not on fork (%x)", receipt.ExecutionResult.BlockID)
 		}
 
-		valid, err := IsValidReceipt(receipt)
+		err = m.validator.Validate(receipt)
 		if err != nil {
-			return fmt.Errorf("could not check validity of Execution Receipt %v: %w", receipt.ID(), err)
-		}
-		if !valid {
-			return state.NewInvalidExtensionErrorf("payload includes invalid receipt (%x)", receipt.ID())
+			return fmt.Errorf("payload includes invalid receipt %v: %w", receipt.ID(), err)
```

Comment:
I would like to avoid using `(bool, error)` because I agree that it's not informative. I am a fan of returning error if something goes wrong, using sentinel values or `nil` if everything went good. I like your proposal and will do that way.